contrib/cacher2tink/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/tinkerbell/boots/packet"
+)
+
+func main() {
+	var m map[string]interface{}
+	err := json.NewDecoder(os.Stdin).Decode(&m)
+	if err != nil {
+		panic(err)
+	}
+	instance := m["instance"].(map[string]interface{})
```

Comment:
if `m["instance"]` is nil, the type assertion will cause a panic. might want to check if `m["instance"]` is not nil?

Reply:
good calls (here and metadata). I don't think instance will actually ever be nil the way PAPI sends the data, but doesn't hurt to be defensive.

Reply:
fixed

Reply:
this will still panic if `m["instance"] == nil` https://goplay.tools/snippet/nzhYbavY_sS
one option could be to change line 13 from `var m map[string]interface{}` to `m := make(map[string]interface{})`

Reply:
ah yeah, if stdin is `{}`, good call. I'll fix that too.

Reply:
fixed, https://goplay.tools/snippet/6BK3lENcTm9